### DISCUSSION TITLE
Dicas para quem quer criar seu próprio Framework (indiferente da Linguagem)

### DISCUSSION THREAD

**Discussion by daniloagostinho** [frontendbr/forum#521] [Perguntas e Respostas]
Boa noite pessoal, ;)

Quais são as dicas de vocês pra quem deseja criar o próprio Framework? Na internet existem milhares, mas, eu gostaria de consolidar meu conhecimento em **Frontend**.  E criar algo, pode ser um **Bootstrap** da vida, ou uma lib.. **JS**. 

Vlw's

**Comment by bernardodiasc** [comment]
Postei esse video na #447, acho que tambem é relacionado com a sua questão: https://www.youtube.com/watch?v=T1YWB9N166A Browser Framework Fundamentals - Yoshua Wuyts

**Comment by daniloagostinho** [comment]
@bernardodiasc vou dar uma olhada. :+1:

**Comment by nandomoreirame** [comment]
Tem alguns artigos no Tableless bem interessantes com esse tema:
- https://tableless.com.br/criando-seu-framework-html-css/
- https://tableless.com.br/biblioteca-css-ou-framework/
- https://tableless.com.br/locaweb-style-como-iniciamos/
- https://tableless.com.br/estruturacao-de-client-side-preprocessadores-framewoks-e-bibliotecas-parte-1/

Enjoy!

**Comment by mahenrique94** [comment]
@DaniloAgostinho o primeiro passo seria criar um padrão de nomeclatura das suas classes, seja ele BEM, SMACSS, entre outros.

Nesse link o tableless explica a diferença dos padrões de CSS: [https://tableless.com.br/oocss-smacss-bem-dry-css-afinal-como-escrever-css/](https://tableless.com.br/oocss-smacss-bem-dry-css-afinal-como-escrever-css/).

Talvez ler um livro de CSS sobre seus padrões seria interessante também: [https://www.casadocodigo.com.br/products/livro-css-eficiente](https://www.casadocodigo.com.br/products/livro-css-eficiente).

Um outro detalhe, seria legal você utilizar um pré processador para gerar seu CSS final, seu código ficará mais organizado e fácil de manutenção, o [SASS](https://www.casadocodigo.com.br/products/livro-sass) é um dos grandes famosos no quesito.

O resto é simplesmente ir fazendo seus componentes e boa.

**Comment by fernandosouza** [comment]
Você teria alguma necessidade prática a ser resolvida? Gostaria que um determinado FW que você usa tivesse uma arquitetura diferente ou acha que poderia fazer melhorias?

Indentificar pontos como esse podem ser o primeiro passo. Se você souber o que quer implementar e para quem você quer implementar, todas as outras decisões tecnicas podem ficar mais fáceis de se resolver.

Você pode identificar necessidades da empresa que você trabalha/trabalhou, necessidades suas em projetos pessoais que fez ou em freelas. Outras pessoas podem se beneficiar dessas tuas ideias e vai ser muito legal.

Nos infroma assim que começar. Seria muito legal acompanhar a evolução do teu projeto. 😄
